### PR TITLE
Add pull-request title guideline

### LINKED ISSUE #6688: Adopt a pattern for PR titles
In order to facilitate the release process, I wanted to propose a guideline for PR titles. Feedback from @open-telemetry/collector-contrib-approvers is appreciated.

There are currently two main patterns I'm seeing related to PR titles, in addition to the "free format" that most are following. The two patterns are:

1. `componentnametype`: description
1. [type/componentname] description

Examples:
1. `lokiexporter`: add complete log record to body
1. [exporter/loki] add complete log record to body

I personally prefer the first option and would like to make it a standard, enforced by approvers and maintainers during the PR review.

If @open-telemetry/collector-contrib-approvers agree with this proposal, I'll send in a PR changing the contributing and reviewing guidelines.

### PR DESCRIPTION
**Pull request by jpkrohling**
Closes #6688 by adding a new guideline for PR titles.

The first commit in this PR simply rewraps the file, as the wrapping wasn't consistent before. It's now wrapped at 120 characters.

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]
I like the 120 character length. It fits the Github split view perfectly.

I think we should enforce it throughout the project for Markdown files

File: receiver/jaegerreceiver/jaeger_agent_test.go [hunk 1]
```diff
@@ -191,11 +191,20 @@ func testJaegerAgent(t *testing.T, agentEndpoint string, receiverConfig *configu
...
+	var err error
+	for i := 0; i < 3; i++ {
+		err = jr.Start(context.Background(), componenttest.NewNopHost())
+		if err == nil {
+			break
+		} else {
+			time.Sleep(50 * time.Millisecond)
+		}
+	}
+	require.NoError(t, err, "Start failed")
```
Comment (dmitryax) [inline receiver/jaegerreceiver/jaeger_agent_test.go:203]: Hm.. didn't catch this. Looks like unrelated?

**Review by codeboten — COMMENTED** [review]
Text change looks good, please revert the test change :)

File: receiver/jaegerreceiver/jaeger_agent_test.go (hunk 1 above)
Comment (jpkrohling) [reply receiver/jaegerreceiver/jaeger_agent_test.go:203]: Oops, sorry, made this branch on top of another PR. Sorry about that.

**Review by codeboten — APPROVED** [review]
